### PR TITLE
libav_encoder: Allow other libav encoders to be selected.

### PR DESCRIPTION
**Pull request by naushir**
Add a general codec options function that will allow other libav encoders to be selected correctly.

### CODE REVIEW

File: encoder/libav_encoder.cpp [hunk 1]
```diff
@@ -20,19 +20,40 @@
...
+void encoderOptionsGeneral(VideoOptions const *options, AVCodecContext *codec)
 {
-	codec->pix_fmt = AV_PIX_FMT_DRM_PRIME;
-	codec->max_b_frames = 0;
+	codec->framerate = { (int)(options->framerate.value_or(DEFAULT_FRAMERATE) * 1000), 1000 };
+
+	if (!options->profile.empty())
+	{
+		static const std::map<std::string, int> profile_map = {
+			{ "baseline", FF_PROFILE_H264_BASELINE },
```
[the remark this answers is not in the document]
Comment (naushir) [reply encoder/libav_encoder.cpp:30]: Good point.  This will have to be moved out of here.

[the remark this answers is not in the document]
Comment (popsUlfr) [reply encoder/libav_encoder.cpp:30]: I haven't tested this myself yet, this should make the profile selection adapt to the encoder:

```diff
diff --git a/encoder/libav_encoder.cpp b/encoder/libav_encoder.cpp
index b69d442..f5a77e8 100644
--- a/encoder/libav_encoder.cpp
+++ b/encoder/libav_encoder.cpp
@@ -23,20 +23,22 @@ namespace {
 void encoderOptionsGeneral(VideoOptions const *options, AVCodecContext *codec)
 {
 	codec->framerate = { (int)(options->framerate.value_or(DEFAULT_FRAMERATE) * 1000), 1000 };
+	codec->profile = FF_PROFILE_UNKNOWN;
 
 	if (!options->profile.empty())
 	{
-		static const std::map<std::string, int> profile_map = {
-			{ "baseline", FF_PROFILE_H264_BASELINE },
-			{ "main", FF_PROFILE_H264_MAIN },
-			{ "high", FF_PROFILE_H264_HIGH }
-		};
-
-		auto it = profile_map.find(options->profile);
-		if (it == profile_map.end())
+		const AVCodec *encoder_codec = avcodec_find_encoder_by_name(options->libav_video_codec.c_str());
+		for (const AVProfile *encoder_profile = encoder_codec->profiles;
+			encoder_profile && encoder_profile->profile != FF_PROFILE_UNKNOWN;
+			encoder_profile++) {
+			if (!options->profile.compare(encoder_profile->name))
+			{
+				codec->profile = encoder_profile->profile;
+				break;
+			}
+		}
+		if (codec->profile == FF_PROFILE_UNKNOWN)
 			throw std::runtime_error("libav: no such profile " + options->profile);
-
-		codec->profile = it->second;
 	}
 
 	codec->level = options->level.empty() ? FF_LEVEL_UNKNOWN : std::stof(options->level) * 10;
```

[the remark this answers is not in the document]
Comment (naushir) [reply encoder/libav_encoder.cpp:30]: Excellent!  I will update this change to match your code.

**Comment by naushir** [comment]
Moved to https://github.com/raspberrypi/rpicam-apps/pull/606 to get the runners working.

**Pull request closed without merging**